stable_baselines/common/vec_env/vec_normalize.py

```diff
@@ -37,6 +41,45 @@ def __init__(self, venv, training=True, norm_obs=True, norm_reward=True,
         self.norm_reward = norm_reward
         self.old_obs = np.array([])
 
+    def __getstate__(self):
+        """
+        Gets state for pickling.
+
+        Excludes self.venv, as in general VecEnv's may not be pickleable."""
+        state = self.__dict__.copy()
+        # these attributes are not pickleable
+        del state['venv']
+        del state['class_attributes']
+        # these attributes depend on the above and so we would prefer not to pickle
+        del state['ret']
+        return state
+
+    def __setstate__(self, state):
+        """
+        Restores pickled state.
+
+        User must call set_venv() after unpickling before using.
+
+        :param state: (dict)"""
+        self.__dict__.update(state)
+        assert 'venv' not in state
+        self.venv = None
+
+    def set_venv(self, venv):
```

Comment:
Now that the `venv` is mandatory for loading, does this method still makes sense?

Because `if self.venv is not None:` will never be False, no?

Reply:
I'm imagining `pickle.dump()`/`pickle.load()` still being part of the public API, but not being the normal use case. If you just do `pickle.load()` then `venv` will be `None`. I don't have strong feelings on this though -- am OK removing the method.